services/symfony/symfony-insight.tester.js

```diff
@@ -0,0 +1,316 @@
+'use strict'
+
+const Joi = require('joi')
+const { colorScheme } = require('../test-helpers')
+const t = (module.exports = require('../create-service-tester')())
+const { withRegex } = require('../test-validators')
+
+const {
+  runningMockResponse,
+  platinumMockResponse,
+  goldMockResponse,
+  silverMockResponse,
+  bronzeMockResponse,
+  noMedalMockResponse,
+  mockSymfonyUser,
+  mockSymfonyToken,
+  mockSymfonyInsightCreds,
+  restore,
+  realTokenExists,
+  prepLiveTest,
+  criticalViolation,
+  majorViolation,
+  minorViolation,
+  infoViolation,
+  multipleViolations,
+} = require('./symfony-test-helpers')
+
+const sampleProjectUuid = '45afb680-d4e6-4e66-93ea-bcfa79eb8a87'
+
+beforeEach(function() {
+  mockSymfonyInsightCreds()
```

Comment:
I was hoping to be able to leverage mocha's `describe/context` to separate the different types of tests with their own respective setup methods, but it seems that doesn't work (I could only get top level `before`/`beforeEach` to work).

As such I figured I'd either have to add `.before(..)` and `.finally(..)` calls on each individual test, or use this approach with a singular `beforeEach` and `afterEach` function. This did require a tweak on the checks used in the live tests to ensure that if real tokens existed they would be used, noted below

Reply:
Ah, gotcha. In IcedFrisby you can wrap calls to `frisby.create().toss()` inside an `it()` inside a `describe()` block, and IIRC Mocha will end up running that `describe`'s before hooks. It won't work here because `toss()` doesn't get invoked until well after all the `t.create()` calls have run, which is too late. If you invoke `toss()` directly, the rest of the service test runner won't get a chance to do its thing (e.g. limit the runner to running what we want in a particular PR). (The `beforeEach` hook you've added here works because it's global: it applies to _every_ service test, not just the ones defined in this file. )

You could work around it like this:

```js
function createWithMockCreds(title) {
  return t.create(title)
    .before(mockSymfonyInsightCreds)
    .finally(...)
}
```

or

```js
function create(title, { withMockCreds = false } = {}) {
  const result = t.create(title)
  if (withMockCreds) {
    result.before(...)
    result.finally(...)
  }
  return result
}

Reply:
> The beforeEach hook you've added here works because it's global: it applies to every service test, not just the ones defined in this file.

Yikes!

Reply:
I am borderline asleep ha so just to make sure I am following correctly

I can add one of those functions, and then update the test calls to use that function instead of `t.create(...`) correct? For example

```js
t.create('platinum grade')
  .get(`/symfony/i/grade/${sampleProjectUuid}.json?style=_shields_test`)
  .intercept(nock =>
    nock('https://insight.symfony.com/api/projects')
      .get(`/${sampleProjectUuid}`)
      .reply(200, platinumMockResponse)
  )
  .expectJSON({
    name: 'grade',
    value: 'platinum',
    colorB: '#E5E4E2',
  })
```

would become: 


```js
create('platinum grade', { withMockCreds: true })
  .get(`/symfony/i/grade/${sampleProjectUuid}.json?style=_shields_test`)
  .intercept(nock =>
    nock('https://insight.symfony.com/api/projects')
      .get(`/${sampleProjectUuid}`)
      .reply(200, platinumMockResponse)
  )
  .expectJSON({
    name: 'grade',
    value: 'platinum',
    colorB: '#E5E4E2',
  })
```

Reply:
Yup, exactly!